In [9]:
from scipy import stats
from scipy.stats import chi2_contingency
import pandas as pd
import numpy as np

In [16]:
from google.colab import files
uploaded = files.upload()

Saving Cutlets.csv to Cutlets.csv


In [17]:
data = pd.read_csv("Cutlets.csv")

In [18]:
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [19]:
A= data["Unit A"]
A.head()

0    6.8090
1    6.4376
2    6.9157
3    7.3012
4    7.4488
Name: Unit A, dtype: float64

In [20]:
B= data["Unit B"]
B.tail()

30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

Q1) Ans:
Assumptions: **1)Assume Null hyposthesis as Ho: A = B (There is no difference in diameters of cutlets between two units).**

**2)Thus Alternate hypothesis as Ha: A ≠ B (There is significant difference in diameters of cutlets between two units)**

In [ ]:
Cutlet = stats.ttest_ind(A,B)              ##### 2 Sample 2 Tail test applicable
Cutlet

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

In [ ]:
Cutlet[1]

0.4722394724599501

 **Result**=
 Here p_value(0.4722)>aplha(Significance Value,0.05) hence **rejecting null hypothesis.**
 So their is **no significant difference** between diameters of two cutlets from different units.

**Q2)Ans**
Assume **Null hyposthesis as Ho: Their is no difference between average Turn Around Time
Alternate hypothesis as H1: Their is difference between average Turn Around Time**

In [ ]:
from google.colab import files
uploaded= files.upload()

Saving LabTAT.csv to LabTAT.csv


In [ ]:
Lab= pd.read_csv("LabTAT.csv")
Lab.head(5)

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
p_value = stats.f_oneway(Lab['Laboratory 1'],Lab['Laboratory 2'],Lab['Laboratory 3'],Lab['Laboratory 4'])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [ ]:
p_value[1]

2.1156708949992414e-57

Here p_value(2.115e-57)<aplha(Significance Value,0.05) hence **rejecting null
hypothesis**.
So their is **significant difference** between average Turn Around Time.

**Q3)Ans:**
Assumptions: **a) Null Hypothesis Ho= All the proporations are equal.
             b) Alternate Hypothesis H1= All the proporations are not equal.
             c) alpha value= 0.05**

In [2]:
from google.colab import files
uploaded = files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [3]:
Sales= pd.read_csv("BuyerRatio.csv")
Sales

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [13]:
a= Sales.iloc[:,1:6]
a

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [14]:
stats.chi2_contingency(a)     ####converting observed values into expected

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [5]:
observed = np.array([50, 142, 131, 70, 435, 1523, 1356, 750])
expected = np.array([42.76531299,  146.81287862,  131.11756787, 72.30424052, 442.23468701, 1518.18712138, 1355.88243213, 747.69575948])

In [21]:
stats.chisquare(observed, expected, ddof = 3)

Power_divergenceResult(statistic=1.5959455390914483, pvalue=0.8095206646905712)

Here p_value(0.809)>aplha(Significance Value,0.05) hence **accepting null
hypothesis**.
So their is ** no significant difference** between ratios of male and females

**Q4) Ans:**
Assumptions:**a)Ho(Null Hypothesis)= The defective% doesn't vary by centers.
            b)H1(Alternate Hypothesis)= The defective% vary by centers.**

In [23]:
from google.colab import files
uploaded = files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [24]:
centers = pd.read_csv("Costomer+OrderForm.csv")
centers

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [41]:
z=print(centers['Phillippines'].value_counts(),"\n",centers['Indonesia'].value_counts(),"\n",centers['Malta'].value_counts(),"\n",centers['India'].value_counts())
z

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 
 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 
 Error Free    269
Defective      31
Name: Malta, dtype: int64 
 Error Free    280
Defective      20
Name: India, dtype: int64


In [46]:
observed = [(271,267,269,280),(29,33,31,20)]

In [48]:
prob = stats.chi2_contingency(observed)
prob

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

In [49]:
prob[1]

0.2771020991233135

Here p_value(0.2771)>aplha(Significance Value,0.05) hence **accepting null
hypothesis**.
So their is ** no significant difference** between defective% of centers.